src/chain.h

```diff
@@ -263,7 +263,7 @@ class CBlockIndex
 
     uint256 GetBlockHash() const
     {
-        return *phashBlock;
+        return *Assert(phashBlock);
```

Comment:
I think this should be turned into `assert(phashBlock);` (or even better `assert(phashBlock != nullptr)`). It will crash for sure even in release builds where `Assert()` is a noop. Better provide some meaningful message in that case (from the assert), before crashing. Then this assert can be removed:

https://github.com/bitcoin/bitcoin/blob/895937edb2cca3046640dc016827a216e0b6d1c1/src/validation.cpp#L2269-L2271

Reply:
Done and done, thanks!